intermediate_source/torchserve_with_ipex_2.rst

```diff
@@ -0,0 +1,471 @@
+=====================================================================
+Grokking PyTorch Intel CPU performance from first principles (Part 2)
+=====================================================================
+
+Authors: Min Jean Cho, Jing Xu, Mark Saroufim
+
+In the previous blog, `Grokking PyTorch Intel CPU Performance From First Principles <https://pytorch.org/tutorials/intermediate/torchserve_with_ipex.html>`_
+, we have introduced how to tune CPU runtime configurations, how to profile them, and how to integrate them into `TorchServe <https://github.com/pytorch/serve>`_ for optimized CPU performance.
+
+In this blog, we'll demonstrate boosting performance with memory allocator via the `Intel® Extension for PyTorch* Launcher <https://github.com/intel/intel-extension-for-pytorch/blob/master/docs/tutorials/performance_tuning/launch_script.md>`_
+, and optimized kernels on CPU via `Intel® Extension for PyTorch* <https://github.com/intel/intel-extension-for-pytorch>`_
+, and apply them to TorchServe showcasing 7.71x throughput speedup for ResNet50, and 2.20x throughput speedup for BERT. 
+
+.. figure:: /_static/img/torchserve-ipex-images-2/1.png
+   :width: 100%
+   :align: center
+   
+Throughout this blog, we'll use `Top-down Microarchitecture Analysis (TMA) <https://www.intel.com/content/www/us/en/develop/documentation/vtune-cookbook/top/methodologies/top-down-microarchitecture-analysis-method.html>`_ to profile and show that the Back End Bound (Memory Bound, Core Bound) is often the primary bottleneck for under-optimized or under-tuned deep learning workloads, and we'll demonstrate optimization techniques via Intel® Extension for PyTorch* for improving Back End Bound. We'll also use `Intel® VTune™ Profiler's Instrumentation and Tracing Technology (ITT) <https://github.com/pytorch/pytorch/issues/41001>`_ to profile at finer granularity.
+
+*****************
+Table of Contents
+*****************
+- Prerequisite  
+    - Top-down Microarchitecture Analysis Method (TMA)
+    - Intel® VTune™ Profiler's Instrumentation and Tracing Technology (ITT)
+- TorchServe with Intel® Extension for PyTorch*
+    - Leveraging Advanced Launcher Configuration: Memory Allocator 	
+        - TCMalloc, JeMalloc, PTMalloc
+        - Exercise
+    - Intel® Extension for PyTorch*
+        - Intel® Extension for PyTorch* Optimizations  
+        - Intel® Extension for PyTorch* with TorchServe 
+        - Exercise
+        
```

Comment:
Where is prerequisites? In the TOC above they are listed as first section.

Reply:
Do we need to have an explicit heading with "Prerequisites" ? The TOC is more of intended for a glance of content at one shot, rather than a link to the sections.

Reply:
It would be good to have a prerequisites section and list what the user need to know / have installed before they can complete this tutorial.

Reply:
OK. Let me try moving around some paragraphs, to heave content under each heading.